wgpu-hal/src/gles/egl.rs

```diff
@@ -986,27 +1021,35 @@ impl crate::Surface<super::Api> for Surface {
                 attributes.push(egl::ATTRIB_NONE as i32);
 
                 // Careful, we can still be in 1.4 version even if `upcast` succeeds
-                let raw_result = match self.egl.upcast::<egl::EGL1_5>() {
-                    Some(egl) if self.wsi.kind != WindowKind::Unknown => {
-                        let attributes_usize = attributes
-                            .into_iter()
-                            .map(|v| v as usize)
-                            .collect::<Vec<_>>();
-                        egl.create_platform_window_surface(
-                            self.display,
-                            self.config,
-                            native_window_ptr,
-                            &attributes_usize,
-                        )
-                    }
-                    _ => self.egl.create_window_surface(
+                #[cfg(not(target_os = "emscripten"))]
```

Comment:
no need for this cfg, it will go the regular `create_window_surface` way naturally, since `self.wsi.kind == WindowKind::Unknown`